spec/ordinary-and-exotic-object-behaviours.html

```diff
@@ -49,14 +46,9 @@ <h1>[[DefineOwnProperty]] ( _P_, _Desc_ )</h1>
         <emu-alg>
           1. Assert: IsPropertyKey(_P_) is *true*.
           1. If Type(_P_) is a Symbol, return *false*.
-          1. If _Desc_.[[Writable]] is present and has value *true*, return *false*.
-          1. If _Desc_.[[Enumerable]] is present and has value *false*, return *false*.
-          1. If _Desc_.[[Configurable]] is present and has value *true*, return *false*.
           1. If IsAccessorDescriptor(_Desc_) is *true*, return *false*.
-          1. Let _current_ be ! RecordGet(_R_, _P_).
-          1. If _current_ is ~empty~ return *false*.
-          1. If _Desc_.[[Value]] is present, return ! SameValue(_Desc_.[[Value]], _current_).
-          1. Return *true*.
+          1. Let _current_ be ! RecordGetOwnProperty(_R_, _P_).
+          1. Return IsCompatiblePropertyDescriptor (*false*, _Desc_, _current_).
```

Comment:
```suggestion
          1. Return IsCompatiblePropertyDescriptor(*false*, _Desc_, _current_).
```

Reply:
This line doesn't check for when `RecordGetOwnProperty` returns `undefined` because the result is `~empty~` - and if you do check for that, you can remove the "if symbol" check above as well.

Reply:
> This line doesn't check for when RecordGetOwnProperty returns undefined because the result is ~empty~ 

Does `undefined` get converted to `~empty~` when returned by an AO?

Reply:
No - what i meant was, inside the implementation of `RecordGetOwnProperty`, if `RecordGet` returns `~empty~`, then `RecordGetOwnProperty` returns `*undefined*`, which isn't checked for here.

Reply:
Ah right, it does look like ValidateAndApplyPropertyDescriptor accepts `undefined` as the current descriptor. Maybe `IsCompatiblePropertyDescriptor` should have it's type updated to also allow `undefined`?

Reply:
That's an option, but it seems simpler to me to have this AO add an "if undefined, return false" check here before the IsCompatiblePropertyDescriptor call

Reply:
Cool. Is the `IsAccessorDescriptor` check _needed_ or was it only an early return. It looks like the result would be the same without it but hard to be sure as `ValidateAndApplyPropertyDescriptor` has a lot of `if`s :D

Reply:
only an early return - but i think the early returns add clarity, even if strictly not necessary

Reply:
thanks for confirming. I'll keep the symbol check as feels like a nice early return too.